windows-gaming-driver/src/setup/wizard.rs

```diff
@@ -249,12 +325,62 @@ impl<'a> Wizard<'a> {
             println!("");
         }
 
-        // TODO: Add a udev picker step here (just like the one we have for gpu) that selects USB keyboard and mouse.
-        // Then save that to the config and create a udev rule file that runs setfacl so the vfio group can use them.
+        // select mouse and keyboard
+        println!("Step 2: Select USB Devices");
+        println!("To enable input devices we use USB passthrough triggered by keyboard shortcuts.");
+        println!();
+        if !machine.usb_devices.is_empty() {
+            println!("You have currently selected the following usb devices: ");
+            for &(vendor, product) in machine.usb_devices.iter() {
+                println!("    {:04x}:{:04x}", vendor, product);
+            }
+            if ask::yesno(&mut self.stdin, "Do you want to remove them before proceeding?") {
+                machine.usb_devices.clear();
+                println!("Removed.");
+            }
+        }
+        println!("Please select your mouse:");
+        let mouse = self.udev_pick_usb(Some(HidKind::Mouse), &machine.usb_devices, true).expect("Failed to select Mouse");
+        println!("Please select your keyboard:");
+        let keyboard = self.udev_pick_usb(Some(HidKind::Keyboard), &machine.usb_devices, true).expect("Failed to select Keyboard");
+        if let Some(id) = mouse {
+            machine.usb_devices.insert(0, id);
+        } else {
+            println!("No mouse selected. Please select your pointing device:");
+            let mouse = self.udev_pick_usb(None, &machine.usb_devices, !machine.usb_devices.is_empty()).expect("Failed to select Pointing Device");
+            if let Some(id) = mouse {
+                machine.usb_devices.insert(0, id);
+            }
+        }
+        if let Some(id) = keyboard {
+            machine.usb_devices.push(id);
+        } else {
+            println!("No keyboard selected. Please select your keyboard:");
+            let keyboard = self.udev_pick_usb(None, &machine.usb_devices, true).expect("Failed to select keyboard from full list");
+            if let Some(id) = keyboard {
+                machine.usb_devices.push(id);
+            }
+        }
+        // add udev rule to add selected devices to vfio group
+        sudo_write_file("/etc/udev/rules.d/80-vfio-usb.rules", |mut w| {
+            for &(vendor, product) in machine.usb_devices.iter() {
+                writeln!(w, r#"SUBSYSTEM=="usb", ATTR{{idVendor}}=="{:04x}", ATTR{{idProduct}}=="{:04x}", ACTION=="add", RUN+="/usr/bin/setfacl -m g:vfio:rw- $devnode""#, vendor, product)?;
+            }
+            Ok(())
+        }).expect("Cannot write udev rules");
+        println!("To passthrough these usb devices you need to execute the following command:");
+        println!("    echo -ne \"\\x01\" | ncat -U /run/user/1000/windows-gaming-driver/control.sock");
```

Comment:
no

why bother doing something that is wrong

and is it really so hard to pack this up into a shell script?

plus telling them *here* is once again **totally wrong**

do that when the setup is down

i wont fucking write down some random piece of trivia that you throw at me in the middle of this wizard when i really don't care